### PR TITLE
[BUG] No initial attachment journal for messages

### PR DESCRIPTION
**Pull request by hschink**
Implements [ticket 2464](https://www.openproject.org/rb/issue_boxes/2464)

:exclamation: Depends on #512

### CODE REVIEW

**Comment by ulferts** [comment]
There is a syntax error

<pre>
spec/controllers/messages_controller_spec.rb:142
</pre>

File: app/models/message.rb [hunk 1]
```diff
@@ -132,4 +133,12 @@ def add_author_as_watcher
     watchers(true)
     watcher_users(true)
   end
+
+  # ACTS AS ATTACHABLE
+  def attachments_changed(obj)
```
Comment (ulferts) [inline app/models/message.rb]: this one is copied over 1:1. How about creating a module for concerns that are attachment + journal related.

**Comment by ulferts** [comment]
I will postpone working on this until #512 is merged

**Comment by hschink** [comment]
Rebased PR

**Comment by tilt** [comment]
Looks good now imo. /cc @ulferts

**Pull request merged**